crates/sui-graphql-rpc/src/types/balance.rs

```diff
@@ -160,3 +205,54 @@ impl TryFrom<StoredBalance> for Balance {
         })
     }
 }
+
+pub(crate) fn build_candidates(snapshot_objs: RawQuery, history_objs: RawQuery) -> RawQuery {
+    let candidates = query!(
+        r#"SELECT DISTINCT ON (object_id) * FROM (
+        ({})
+        UNION
+        ({})
+    ) o"#,
```

Comment:
nit: this could now be one line?

```suggestion
        r#"SELECT DISTINCT ON (object_id) * FROM (({}) UNION ({})) o"#,
```